substrate/frame/revive/src/lib.rs

```diff
@@ -1591,11 +1779,40 @@ impl<T: Config> Pallet<T> {
 		<T::Currency as Inspect<AccountIdOf<T>>>::minimum_balance()
 	}
 
-	/// Deposit a pallet revive event.
+	/// Deposit a pallet contracts event.
+	///
+	/// This method will be called by the EVM to deposit events emitted by the contract.
+	/// Therefore all events must be contract emitted events.
 	fn deposit_event(event: Event<T>) {
+		if matches!(event, Event::ContractEmitted { .. }) &&
+			block_storage::is_executing_ethereum_call()
+		{
+			// TODO: ensure we don't exceed a maximum number of events per tx.
+			InflightEvents::<T>::mutate(|events| {
+				events.push(event.clone());
+			});
+		}
+
 		<frame_system::Pallet<T>>::deposit_event(<T as Config>::RuntimeEvent::from(event))
 	}
 
+	/// Store a transaction payload with extra details.
+	///
+	/// The data is used during the `on_finalize` hook to reconstruct the ETH block.
+	fn store_transaction(payload: Vec<u8>, success: bool, gas_consumed: Weight) {
+		// Collect inflight events emitted by this EVM transaction.
+		let events = InflightEvents::<T>::take();
```

Comment:
ha we reset it here so maybe we don't need to kill on initialize
but I think we still need the kill to ensure that it's not added to the PoV

Reply:
to be honest I don't know maybe @skunert  would know better,
my intuition is that if the storage is killed before anything else, then it's not added to the PoV as you don't ever need to read any existing state